### PR TITLE
chore(common/core): Linux side of caps-lock stores 🚧

### PR DESCRIPTION
**Pull request by ermshiperete**
- implement Linux side of cap-lock action processing
- Turn off capslock on key down
    
   This change improves capslock handling with a capsAlwaysOff keyboard. Previously, the caps lock indicator was turned on while 
   holding down the Caps Lock key. With this change we turn it off immediately. This is done by forcing caps lock off even if the
   state already seems to match.
    
-  Also, for capsAlwaysOff keyboards skip further processing of caps lock key after we turned capslock off again.

Implements part of #5052.

### CODE REVIEW

File: linux/ibus-keyman/src/engine.c [hunk 1]
```diff
@@ -27,6 +27,16 @@
...
 #include <gio/gio.h>
+
+#ifdef GDK_WINDOWING_X11
+#include <gdk/gdkx.h>
+#include <X11/XKBlib.h>
+#endif
+
+#ifdef GDK_WINDOWING_WAYLAND
+#include <gdk/gdkwayland.h>
+#endif
```
Comment (mcdurdin) [inline linux/ibus-keyman/src/engine.c:38]: This seems to be mixing wayland support into this PR? Is it possible to leave out all new wayland-related code and put that into a separate PR?

Do we see scenarios where both wayland and x11 need to be supported simultaneously?

File: linux/ibus-keyman/src/engine.c [hunk 2]
```diff
@@ -238,32 +253,39 @@ static void reset_context(IBusEngine *engine)
...
+    keyman->xdisplay = GDK_DISPLAY_XDISPLAY(keyman->display);
+  } else
+#endif
+#ifdef GDK_WINDOWING_WAYLAND
+  if (GDK_IS_WAYLAND_DISPLAY(keyman->display)) {
+  } else
+#endif
+  {
+    g_error("unsupported GDK backend");
+  }
```
Comment (mcdurdin) [inline linux/ibus-keyman/src/engine.c:280]: I take it this somewhat awkward construct is quite deliberate -- to allow for any permutation of `#define GDK_WINDOWING_X11` and `GDK_WINDOWING_WAYLAND` (with X11 taking precedence over Wayland if both are defined)?

**Review by mcdurdin — APPROVED** [review]
LGTM. I am not sure about including the wayland `#ifdefs` but they are probably okay...?

File: linux/ibus-keyman/src/engine.c (hunk 1 above)
Comment (ermshiperete) [reply linux/ibus-keyman/src/engine.c:38]: > Do we see scenarios where both wayland and x11 need to be supported simultaneously?

yes, the way current Ubuntu versions are configured is that it they build with Wayland, but also have the X11 support enabled for apps that don't support Wayland yet. And I think this will continue for several more years...

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.95-alpha](https://keyman.com/downloads/releases/alpha/15.0.95)
